src/neuroconv/basedatainterface.py

```diff
@@ -46,6 +47,78 @@ def get_metadata(self):
 
         return metadata
 
+    @abstractmethod
+    def get_timestamps(self) -> np.ndarray:
+        """
+        Retrieve the timestamps for the data in this interface.
+
+        Returns
+        -------
+        timestamps: numpy.ndarray
+            The timestamps for the data stream.
+        """
+        raise NotImplementedError(
+            "Unable to retrieve timestamps for this interface! Define the `get_timestamps` method for this interface."
+        )
+
+    @abstractmethod
+    def align_starting_time(self, starting_time: float):
+        """
+        Align the starting time for this interface relative to the common session start time.
+
+        Must be in units seconds relative to the common 'session_start_time'.
+
+        Parameters
+        ----------
+        starting_time : float
+            The starting time for all temporal data in this interface.
+        """
+        raise NotImplementedError(
+            "The protocol for synchronizing the start time of this interface has not been specified!"
+        )
+
+    @abstractmethod
+    def align_timestamps(self, aligned_timestamps: np.ndarray):
+        """
+        Replace all timestamps for this interface with those aligned to the common session start time.
+
+        Must be in units seconds relative to the common 'session_start_time'.
+
+        Parameters
+        ----------
+        aligned_timestamps : numpy.ndarray
+            The synchronized timestamps for data in this interface.
+        """
+        raise NotImplementedError(
+            "The protocol for synchronizing the timestamps of this interface has not been specified!"
+        )
+
+    @abstractmethod
+    def align_by_interpolation(self, aligned_timestamps: np.ndarray, unaligned_timestamps: np.ndarray):
+        """
+        Interpolate the timestamps of this interface using a mapping from some unaligned time basis to its aligned one.
+
+        Use this method if the unaligned timestamps of the data in this interface are not directly tracked by a primary
+        system, but are known to occur between timestamps that are tracked, then align the timestamps of this interface
+        by interpolating between the two.
+
+        An example could be a metronomic TTL pulse (e.g., every second) from a secondary data stream to the primary
+        timing system; if the time references of this interface are recorded within the relative time of the secondary
+        data stream, then their exact time in the primary system is inferred given the pulse times.
+
+        Must be in units seconds relative to the common 'session_start_time'.
```

Comment:
@alejoe91 As someone that often deals with this kind of stuff in experiments, how well does this explanation read for you? Any suggestions?

There will be a follow-up to this PR that focuses exclusively on documentation with some real examples, would you be OK with looking at that when the time comes?